### ISSUE TITLE
Error when using "Returning Into" differs from when it is not been used

### ISSUE THREAD

**Issue by roger-russel** [sijms/go-ora#307]
The syntax without "returning" returns the constraint error, which is the correct problem with the insert.

```
stmt, _ := conn.Prepare(`INSERT INTO foo ( name ) VALUES (:1)`)
if _, err := stmt.Exec("boo"); err != nil {
        fmt.Println(err) // ORA-00001: unique constraint ( uq_foo ) violated
`}`
```

The syntax with "returning" returns a TCC error.
```
stmt, _ := conn.Prepare(`INSERT INTO foo ( name ) VALUES (:1) RETURNING id INTO :2`)
var id int64
if _, err := stmt.Exec("boo",  sql.Out{Dest: &id}); err != nil {
        fmt.Println(err) // TCC error: received code 19 during response reading
`}`
```

I have no idea what this TCC error means, but it looks like it is trying to take the ID without checking if the insert was successful, I didn't check the code it is just my own suspicion, I will try to check how the code does to take the last ID and see what is happening over there.

**Comment by sijms** [comment]
this is because insertion error
no problem. would you share trace log to fix this issue?

**Comment by roger-russel** [comment]
I sent you the logs by email.

**Comment by sijms** [comment]
ok I will try to simulate the condition and find a solution

**Comment by sijms** [comment]
fixed in last release v2.5.32
